test/support/policy_helpers.h

```diff
@@ -0,0 +1,47 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+#ifndef _POLICY_HELPERS_H
+#define _POLICY_HELPERS_H
+
+#include <oneapi/dpl/pstl/execution_defs.h>
+
+#if TEST_DPCPP_BACKEND_PRESENT
+#include <oneapi/dpl/pstl/hetero/dpcpp/execution_sycl_defs.h>
+#endif
+
+#include <type_traits>
+
+namespace TestUtils
+{
+template <class _ExecPolicy, const bool __condition = false, class _T = void>
+using enable_if_host_execution_policy_conditional_or = ::std::enable_if_t<
+    oneapi::dpl::__internal::__is_host_execution_policy<::std::decay_t<_ExecPolicy>>::value || __condition, _T>;
+
+#if TEST_DPCPP_BACKEND_PRESENT
+template <typename _ExecPolicy, const bool __condition, typename _T = void>
+using enable_if_hetero_execution_policy_conditional = ::std::enable_if_t<
+    oneapi::dpl::__internal::__is_hetero_execution_policy_v<::std::decay_t<_ExecPolicy>> && __condition, _T>;
+#endif
+
+template <class _ExecPolicy, const bool __condition>
+using enable_if_non_hetero_execution_policy_conditional = ::std::enable_if_t<
+#if TEST_DPCPP_BACKEND_PRESENT
+    !oneapi::dpl::__internal::__is_hetero_execution_policy_v<::std::decay_t<_ExecPolicy>> &&
+#endif
+    __condition>;
```

Comment:
While these helpers replace similar constructs in several tests, still they use implementation details of the library. Moreover, combining the policy type check with another condition makes tests less self-explanatory. I prefer to add a single helper that checks if a policy is "heterogeneous" (read: requires the SYCL backend) or not, and only using the public API.

Reply:
Fixed.